packages/framework/client-logger/external-telemetry/src/factory/index.ts

```diff
@@ -0,0 +1,114 @@
+		type: ConsumerConfigType;
+	}
+
+	/**
+	 * Contains constants for the available types for {@link (TelemetryManagerConfig:namespace).IConsumerConfig."type"}
+	 * @beta
+	 */
+	export const ConsumerConfigTypes = {
+		/**
+		 * Identifier to be used for {@link (TelemetryManagerConfig:namespace).IConsumerConfig} of type {@link (TelemetryManagerConfig:namespace).AppInsightsConsumerConfig}
+		 */
+		APP_INSIGHTS: "APP_INSIGHTS",
+	} as const;
+
+	/**
+	 * Aggregate Type for all available types for {@link (TelemetryManagerConfig:namespace).IConsumerConfig."type"}
+	 * @beta
+	 */
+	export type ConsumerConfigType = (typeof ConsumerConfigTypes)[keyof typeof ConsumerConfigTypes];
+
+	/**
+	 * Configuration for using Azure App Insights as the telemetry consumer for a given telemetry manager
+	 * @beta
+	 */
+	export interface AppInsightsConsumerConfig extends IConsumerConfig {
+		type: "APP_INSIGHTS";
+		/** An instance of an Azure Application Insights client {@link @microsoft/applicationinsights-web#ApplicationInsights}
+		 * The instance must be initialed before being used in the config. That command should be  {@link @microsoft/applicationinsights-web#ApplicationInsights.loadAppInsights}
+		 * `applicationInsightsClient.loadAppInsights(); `
+		 */
+		appInsightsClient: ApplicationInsights;
+	}
+}
+
+/**
+ * Config object for creating one or more telemetry managers covering differents scopes of the Fluid Framework
+ *
+ * @beta
+ */
+export interface TelemetryManagerConfig {
+	/**
+	 * Configuration for creating an external telemetry manager for container related telemetry.
+	 */
+	containerTelemetry?: {
+		container: IContainer;
+		consumerConfig: TelemetryManagerConfig.IConsumerConfig;
+	};
+}
+
+/**
+ * This class helps simplify the creation of one or more telemetry managers.
+ */
+export class TelemetryManagerFactory {
+	static createTelemetryManagers(config: TelemetryManagerConfig): {
+		container?: ContainerTelemetryManager;
+	} {
+		let containerTelemetryManager;
+		if (
+			config.containerTelemetry &&
+			config.containerTelemetry.consumerConfig.type ===
+				TelemetryManagerConfig.ConsumerConfigTypes.APP_INSIGHTS
+		) {
+			const consumerConfig = config.containerTelemetry
+				.consumerConfig as TelemetryManagerConfig.AppInsightsConsumerConfig;
+			const container = config.containerTelemetry.container;
+			const telemetryProducer = new ContainerEventTelemetryProducer(container);
+			const telemetryConsumer = new AppInsightsTelemetryConsumer(
+				consumerConfig.appInsightsClient,
+			);
+			containerTelemetryManager = new ContainerTelemetryManager(
+				container,
+				telemetryProducer,
+				telemetryConsumer,
+			);
+		}
+
+		return {
+			container: containerTelemetryManager,
+		};
+	}
+}
+
+/**
+ * Creates external telemetry managers for one or more areas of the Fluid Framework.
+ * @beta
+ */
+export const createTelemetryManagers = (config: TelemetryManagerConfig): void => {
+	TelemetryManagerFactory.createTelemetryManagers(config);
```

Comment:
Just inline the static function from the factory here? We don't really need the factory class for anything. This should probably return the created object though. And I think it doesn't need the `s` at the end, it creates a single telemetry manager.

Reply:
The reason nothing is returned here is to hide the implementation details from the customer. This makes it so customers have no access to the internal classes and I don't have to export the ContainerTelemetryManager class from the index file.

Since there are no functions/attributes that we expect customers to be calling on this class they don't need to anyways

Reply:
Good point. In that case I think I'd just suggest renaming the function to `startTelemetryManager()`. A `create____()` function makes me feel it should return something that I then use or act on. A `start_____()` one feels more fire-and-forget, which I think is the intention here.

Reply:
Updated.